job_engine/connector/kubernetes.py

```diff
@@ -733,6 +733,48 @@ def reboot(self):
             log.debug(f'The result of the reboot action: {reboot_result}')
         return reboot_result
 
+    @property
```

Comment:
I wouldn't define this method as a property as it's rather "heavy". Please remove the decorator and make an explicit call to it where it is used.

Regarding the name. I would suggest deployed_ne_components().

Reply:
Changed to an explicit call and changed the name for clarity